In [67]:
import pandas as pd
from pyspark.sql import functions as F, SparkSession


In [2]:
spark = (
    SparkSession.builder.appName("Project 1")
    .config("spark.sql.repl.eagerEval.enabled", True) 
    .config("spark.sql.parquet.cacheMetadata", "true")
    .config("spark.sql.session.timeZone", "Etc/UTC")
    .config("spark.driver.memory", "4g")
    .getOrCreate()
)

22/09/04 21:27:45 WARN Utils: Your hostname, DESKTOP-3NQ3PQI resolves to a loopback address: 127.0.1.1; using 172.21.150.117 instead (on interface eth0)
22/09/04 21:27:45 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address


Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).


22/09/04 21:27:47 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable


In [23]:
consumer_ids_sdf = spark.read.parquet('../data/tables/consumer_user_details.parquet')
consumer_ids_df = consumer_ids_sdf.toPandas()

consumer_details_df = pd.read_csv('../data/tables/tbl_consumer.csv', delimiter="|")

merchants_sdf = spark.read.parquet('../data/tables/tbl_merchants.parquet')
merchants_df = merchants_sdf.toPandas()

transactions_sdf = spark.read.parquet('../data/tables/transactions_20210228_20210827_snapshot/order_datetime=2021-08-27/')
transactions_df = transactions_sdf.toPandas()

### consumers

In [83]:
consumer_ids_df

user_id  consumer_id
0             1      1195503
1             2       179208
2             3      1194530
3             4       154128
4             5       712975
...         ...          ...
499994   499995      1385608
499995   499996      1466964
499996   499997      1253484
499997   499998       175005
499998   499999       746835

[499999 rows x 2 columns]

In [31]:
consumer_details_df.head(1)

name                     address state  postcode  gender  \
0  Yolanda Williams  413 Haney Gardens Apt. 742    WA      6935  Female   

   consumer_id  
0      1195503

Each consumer id corresponds to a unique consumer. there are 4999999, unique consumers across the dataset
- 8 possible states
- 3167 possible postcodes across aus
- 3 possible genders

In [30]:
print(len(consumer_details_df))
consumer_details_df.nunique()

499999


name           221377
address        499955
state               8
postcode         3167
gender              3
consumer_id    499999
dtype: int64

### merchants

In [29]:
print(len(merchants_df))
merchants_df.head(1)

4026


name                                               tags  \
0  Felis Limited  ((furniture, home furnishings and equipment sh...   

   merchant_abn  
0   10023283211

Each merchant_abn corresponds to a unique merchant. There are 4026 unique merchants each with a unique name

In [34]:
print(len(merchants_df))
merchants_df.nunique()

4026


name            4026
tags            3954
merchant_abn    4026
dtype: int64

### transactions

In [22]:
transactions_df

user_id  merchant_abn  dollar_value  \
0        18478   20985347699     33.485054   
1            1   63290521567      3.160329   
2        18478   27639240514    387.743012   
3            2   42355028515    852.787636   
4        18478   32082195327      0.699817   
...        ...           ...           ...   
23197    18470   29323795999      8.346682   
23198    18471   21359184622     97.760164   
23199    18472   21807339153     54.310599   
23200    18475   41944909975    103.804012   
23201    18477   63290521567     17.826070   

                                   order_id  
0      22db4e40-f2af-44ad-b091-1070cd29808d  
1      e272051c-aa1f-41be-b84c-8254a68ee2bc  
2      28effd83-18b6-42f5-9929-b0133ff6a089  
3      053bda30-c1e7-4d8c-8ca5-bd95b2a78fd6  
4      18992740-84e8-43e3-b152-94589437b473  
...                                     ...  
23197  f35299fa-c9f4-408e-bb86-b9d5fff65a9a  
23198  8ed0f865-9f70-4e20-8266-81904c95de89  
23199  3251ce95-a850-4fbc-98d0-32bce1651d40  
23200  bd52416a-a7f1-4293-ae96-cc4a8392e414  
23201  0aa657d5-c559-41f7-9345-5d2162ad46f5  

[23202 rows x 4 columns]